### ISSUE TITLE
Type links broken in documentation

### ISSUE THREAD

**Issue by mapsam** [mapbox/mapbox-gl-js#3680]
It appears all of our type links (i.e. `([LngLatBoundsLike])`) are broken in our documentation. For example types that are pointing to the `#Map` class currently point to

```
https://www.mapbox.com/mapbox-gl-js/api/#map
```

when they should be pointing to the uppercase version

```
https://www.mapbox.com/mapbox-gl-js/api/#Map
```

Since these are `id` strings, the capitals matter, so the fragments aren't working as expected. This may be related to an upstream documentation.js issue. I will do some digging.

cc @lucaswoj @lyzidiamond @tmcw

**Comment by mapsam** [comment]
Looks like this is a known issue: https://github.com/documentationjs/documentation/issues/525

@tmcw has started some work on a fix here: https://github.com/documentationjs/documentation/pull/537

**Comment by jfirebaugh** [comment]
#3445 is a special case of this.

**Comment by tmcw** [comment]
Yep, if you can test out that branch, go for it... I'd like to finish out that work - it requires a solid chunk of time to think of All The Cases of All The Things That Might Happen.

**Comment by andrewharvey** [comment]
It looks like this is fixed now!

**Comment by lucaswoj** [comment]
Unfortunately this is not fixed in all cases :-(

**Comment by lucaswoj** [comment]
Fixed by #4460 😄

**Comment by andrewharvey** [comment]
Unfortunately I don't think this is completely fixed. In the code some places use `{@link foo}` while others use `[Foo](Foo)`, the latter remains unfixed.

For example at https://www.mapbox.com/mapbox-gl-js/api/#map#fitbounds under options.easing the AnimationsOptions link is broken.

I think we can simply migrate occurrences of `[Foo](Foo)` to `{@link foo}` to fix this, however it appears that documentation.js doesn't support [all the link formatting options from JSDoc](http://usejsdoc.org/tags-inline-link.html#link-formatting). This is an issue as some of the current markdown style links are monospaced and some aren't.

If we want to retain that monospaced formatting then I'm not sure if we can move to `{@link foo}`.

Related to but not the same issue as #4568

**Comment by kronick** [comment]
This seems like a tough one to squash! I'm seeing it broken still on https://www.mapbox.com/mapbox-gl-js/api/#LngLatLike . Anybody familiar with the innards of how we generate docs able to pick this one up?

**Comment by mollymerp** [comment]
fixed this in #4668 – still need to figure out #4568
